smartdashboard/views.py

```diff
@@ -342,3 +343,203 @@ def __init__(
         self.app_view = app_view
         self.ens_view = ens_view
         self.orc_view = orc_view
+
+
+class MemoryView(ViewBase):
+    """View class for memory section of the Database Telemetry page"""
+
+    def __init__(self, shard: t.Optional[Shard]) -> None:
+        """Initialize a MemoryView
+
+        :param shard: Selected shard
+        :type shard: t.Optional[Shard]
+        """
+        self.shard = shard
+        self.memory_table_element = DeltaGenerator()
+        self.memory_graph_element = DeltaGenerator()
+
+    def update_shard(self, new_shard: t.Optional[Shard]) -> None:
+        """Update selected shard
+
+        This is called after a new shard is selected
+        in the dashboard to keep displayed data in sync.
+
+        :param new_shard: Selected shard
+        :type new_shard: Optional[Shard]
+        """
+        if new_shard is not None:
+            self.shard = new_shard
```

Comment:
A user can't set it to `None` themselves. It can be set to `None` if there aren't any `Orchestrators` or `Orchestrators` with `Shards` (I don't think the latter is possible though?).

Reply:
Now that `MemoryView` can't be set multiple times within `memory_view_builder`, I took out the shard property. Let me know if that was a bad call!